lib/composite_cache_store.rb

```diff
@@ -4,120 +4,132 @@
 require_relative "composite_cache_store/version"
 
 class CompositeCacheStore
-  DEFAULT_OUTER_OPTIONS = {
+  DEFAULT_LAYER_1_OPTIONS = {
     expires_in: 5.minutes,
     size: 16.megabytes
   }
 
-  DEFAULT_INNER_OPTIONS = {
+  DEFAULT_LAYER_2_OPTIONS = {
     expires_in: 1.day,
     size: 32.megabytes
   }
 
-  attr_reader :outer_cache_store, :inner_cache_store
-
-  alias_method :outer, :outer_cache_store
-  alias_method :inner, :inner_cache_store
+  attr_reader :layers
+  delegate_missing_to :"layers.last"
 
   # Returns a new CompositeCacheStore instance
-  # - inner_cache_store: An ActiveSupport::Cache::Store instance to use for the inner cache store (typically remote)
-  # - outer_cache_store: An ActiveSupport::Cache::Store instance to use for the outer cache store (typically local)
-  def initialize(options = {})
-    options ||= {}
-
-    @inner_cache_store = options[:inner_cache_store]
-    @inner_cache_store = ActiveSupport::Cache::MemoryStore.new(DEFAULT_INNER_OPTIONS) unless inner.is_a?(ActiveSupport::Cache::Store)
+  def initialize(*layers)
+    if layers.blank?
+      layers << ActiveSupport::Cache::MemoryStore.new(DEFAULT_LAYER_1_OPTIONS)
+      layers << ActiveSupport::Cache::MemoryStore.new(DEFAULT_LAYER_2_OPTIONS)
+    end
 
-    @outer_cache_store = options[:outer_cache_store]
-    @outer_cache_store = ActiveSupport::Cache::MemoryStore.new(DEFAULT_OUTER_OPTIONS) unless outer.is_a?(ActiveSupport::Cache::Store)
+    layers.freeze
```

Comment:
I wonder if we should still `raise` here, in case any of the `layers` isn't a cache store.